### PR TITLE
HTTP API whitelists do not always work as expected - Closes #4629

### LINKED ISSUE #4629: HTTP API whitelists do not always work as expected
- Setting the `LISK_API_WHITELIST` environment variable (e.g by adding `export LISK_API_WHITELIST=127.0.0.1` to `env.sh`) breaks the whitelist: no IP whatsoever is allowed to connect (in the example `127.0.0.1` will not get an HTTP code `403`
- Similarly setting `LISK_FORGING_WHITELIST` breaks the "forging" whitelist
- Setting `.modules.http_api.forging.access.whiteList[0]` in a custom `config.json` also breaks the "forging" whitelist.

I strongly suspect this behaviour was introduced with `2.0` but I can only state with certainty that is is present in `2.0.2`, `2.1.3` and a recent build from the `development` branch. See this [spreadsheet](https://docs.google.com/spreadsheets/d/1CVndVU-1IWkUXq9hMAUfDRzTXpGLfAiXOEaYZTsh-C0/view) for details.

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
The configuration specification for http_api module were using  `stringToIpPortSet` formatter.

### How did I solve it?

Created new formatter `stringToIpList` and updated the configuration spec for http_api module. 

### How to manually test it?

Added few tests as well, so can be tested with `nom run jest:unit`

### Review checklist

- [ ] The PR resolves #4629
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: framework/test/jest/unit/specs/controller/helpers/validator/keywords/formatters.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+const {
+	stringToIpPortSet,
+	stringToIpList,
+	stringToDelegateList,
+} = require('../../../../../../../../src/controller/validator/keywords/formatters');
+
+describe('formatters', () => {
+	describe('stringToIpPortSet', () => {
+		it('should return empty array if given argument is not string', async () => {
```
Comment (ManuGowda) [inline framework/test/jest/unit/specs/controller/helpers/validator/keywords/formatters.spec.js]: do we need `async` here? same applies to all other cases below.

Comment (nazarhussain) [reply framework/test/jest/unit/specs/controller/helpers/validator/keywords/formatters.spec.js]: Theoretically no need. There were a es-lint check earlier no-sync for mocha tests, so I had habit to have every test async.

File: framework/src/controller/validator/keywords/formatters.js [hunk 2]
```diff
@@ -21,13 +21,20 @@ module.exports = {
 				const [ip, wsPort] = peer.split(':');
 				return {
 					ip,
-					wsPort: wsPort || 5000,
+					wsPort: parseInt(wsPort, 10) || 5000,
```
Comment (fchavant) [inline framework/src/controller/validator/keywords/formatters.js]: Is this fixing another bug you found while fixing the whitelist issue?

**Comment by fchavant** [comment]
re issue number 3

```
wget http://lisk-releases.ams3.digitaloceanspaces.com/lisk-core/lisk-3.0.0-alpha.2-78bf54499-20622d4-Linux-x86_64.tar.gz http://lisk-releases.ams3.digitaloceanspaces.com/lisk-core/lisk-3.0.0-alpha.2-78bf54499-20622d4-Linux-x86_64.tar.gz.SHA256
sha256sum -c lisk-3.0.0-alpha.2-78bf54499-20622d4-Linux-x86_64.tar.gz.SHA256
tar xf lisk-3.0.0-alpha.2-78bf54499-20622d4-Linux-x86_64.tar.gz
cd lisk-3.0.0-alpha.2-78bf54499-20622d4-Linux-x86_64/
./lisk.sh coldstart -0
curl http://127.0.0.1:7000/api/node/status  # 200
curl http://$EXTERNAL_IP:7000/api/node/status  # 403
echo {} >empty.json
jq ".modules.http_api.forging.access.whiteList=[\"$EXTERNAL_IP\"]" empty.json >config.json
./lisk.sh reload
curl http://127.0.0.1:7000/api/node/status/forging  # 403
curl http://$EXTERNAL_IP:7000/api/node/status/forging  # 403
```

Note that both curl calls returns `403`.

```
source env.sh
node scripts/generate_config.js -c config.json |jq .modules.http_api.access.whiteList
[
  "157.245.75.75"
]
```

```
cat config.json
{
  "modules": {
    "http_api": {
      "forging": {
        "access": {
          "whiteList": [
            "157.245.75.75"
          ]
        }
      }
    }
  }
}
```

File: framework/src/controller/validator/keywords/formatters.js (hunk 2 above)
Comment (nazarhussain) [reply framework/src/controller/validator/keywords/formatters.js]: Kind of inconsistency not the issue. Somewhere sport was used as integers and while passed through command line it was considered as string.

**Pull request merged**